### PR TITLE
FuelRouting

### PR DESCRIPTION
**Pull request by matteocrippa**
As proposed in #216 here the related PR for the routing feature.

I have updated also the readme with the new feature and example (I also updated that with your suggestion using fold).

### CODE REVIEW

File: fuel/src/main/kotlin/com/github/kittinunf/fuel/util/FuelRouting.kt [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+     */
+    override val request: Request
+        get() {
+            // generate the encoder according provided parameters, headers, path, etc.
+            val encoder = Encoding().apply {
+                this.baseUrlString = basePath
+                this.httpMethod = method
+                this.urlString = path
+                this.parameters = params
+                // FIXME: headers are missing
```
Comment (kittinunf) [inline fuel/src/main/kotlin/com/github/kittinunf/fuel/util/FuelRouting.kt]: Is there anything we can do for this? What do you mean by `headers` are missing though?

Comment (matteocrippa) [inline fuel/src/main/kotlin/com/github/kittinunf/fuel/util/FuelRouting.kt]: I see there's no way to set headers and inject them like parameters or so. Looking at `encoder` I don't see any particular way to append them to the request

Comment (yoavst) [inline fuel/src/main/kotlin/com/github/kittinunf/fuel/util/FuelRouting.kt]: What's about appending them after you have a `Request` object?

Comment (matteocrippa) [inline fuel/src/main/kotlin/com/github/kittinunf/fuel/util/FuelRouting.kt]: makes sense, I'll update in that way, I was expecting to set it in apply, but I totally miss that

**Comment by kittinunf** [comment]
This looks great to me. If @yoavst say thumb up for this, I am happy to merge this in.

**Comment by yoavst** [comment]
:+1:

**Pull request merged**